src/hotspot/os/linux/os_linux.cpp

```diff
@@ -292,15 +292,18 @@ bool os::Linux::free_memory(size_t& value) {
 }
 
 bool os::total_swap_space(size_t& value) {
-  if (OSContainer::is_containerized() && OSContainer::memory_and_swap_limit_in_bytes() > 0) {
-    if (OSContainer::memory_limit_in_bytes() > 0) {
-      value = static_cast<size_t>(OSContainer::memory_and_swap_limit_in_bytes() - OSContainer::memory_limit_in_bytes());
+  if (OSContainer::is_containerized()) {
+    jlong memory_and_swap_limit_in_bytes = OSContainer::memory_and_swap_limit_in_bytes();
+    jlong memory_limit_in_bytes = OSContainer::memory_limit_in_bytes();
+    if (memory_limit_in_bytes > 0 && memory_and_swap_limit_in_bytes > 0) {
+      value = static_cast<size_t>(memory_and_swap_limit_in_bytes - memory_limit_in_bytes);
       return true;
     }
   } // fallback to the host swap space if the container did return the unbound value of -1
   struct sysinfo si;
   int ret = sysinfo(&si);
   if (ret != 0) {
+    assert(false, "sysinfo failed in total_swap_space()?");
```

Comment:
We should report the value of errno in these cases.
```suggestion
    assert(false, "sysinfo failed in total_swap_space(): %s", os::strerror(errno));
```

Reply:
Added `errno `reporting as suggested.